deepchem/models/torch_models/layers.py

```diff
@@ -2972,3 +2972,65 @@ def _dynamic_partition(self, input_tensor: torch.Tensor,
         partitions = [input_tensor[mask] for mask in partition_masks]
 
         return partitions
+
+
+class DTNNEmbedding(nn.Module):
+    """DTNNEmbedding layer for DTNN model.
+    Assign initial atomic descriptors.
+
+    Parameters
+    ----------
+    n_embedding: int, optional
+        Number of features for each atom
+    periodic_table_length: int, optional
+        Length of embedding, 83=Bi
+    init: str, optional
+        Weight initialization for filters.
+
+    Examples
+    --------
+    >>> from deepchem.models.torch_models import layers
+    >>> import torch
+    >>> layer = layers.DTNNEmbedding(30, 30, 'xavier_uniform_')
+    >>> output = layer(torch.tensor([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]))
+    >>> output.shape
+    torch.Size([10, 30])
+
+    """
+
+    def __init__(self,
+                 n_embedding: int = 30,
+                 periodic_table_length: int = 30,
+                 init: str = 'xavier_uniform_',
+                 **kwargs):
+
+        super(DTNNEmbedding, self).__init__(**kwargs)
+        self.n_embedding = n_embedding
+        self.periodic_table_length = periodic_table_length
+        self.init = init  # Set weight initialization
+
+        init_func: Callable = getattr(initializers, self.init)
+        self.embedding_list: torch.Tensor = init_func(
+            torch.empty([self.periodic_table_length, self.n_embedding]))
+
+    def __repr__(self) -> str:
+        """Returns a string representing the configuration of the layer.
+
+        Returns
+        -------
+        n_embedding: int, optional
+            Number of features for each atom
+        periodic_table_length: int, optional
+            Length of embedding, 83=Bi
+        init: str, optional
+            Weight initialization for filters.
+
+        """
+        return f'{self.__class__.__name__}(n_embedding={self.n_embedding}, periodic_table_length={self.periodic_table_length}, init={self.init})'
+
+    def forward(self, inputs):
+        """
+        parent layers: atom_number
+        """
+        atom_number = inputs
```

Comment:
Needs type hints. What is inputs?

Reply:
I have made some changes here. I think something better can be done here do you have any suggestion.